<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Shashank" />

<meta name="date" content="2021-11-06" />

<title>Chicane Analysis</title>

<script src="site_libs/header-attrs-2.8/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Chicane-analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About Data</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Chicane Analysis</h1>
<h4 class="author">Shashank</h4>
<h4 class="date">11/06/2021</h4>

</div>


<div id="initial-observations" class="section level1">
<h1>Initial Observations</h1>
<p>We look to analyse the ouput from the initial chicane run. Compare the results from the previous chicago run.</p>
</div>
<div id="introduction-to-the-chicane" class="section level1">
<h1>Introduction to the Chicane</h1>
<p>CHiCANE was designed to identify regions of interest that interact more than expected by chance. To indetify the interaction peaks, CHiCANE models the expected number of reads linking two restriction fragments as a function of the distance between the loci and the ‘interactibility’ of the bait fragment; that is, its inherent propensity to interact with other fragments.</p>
<p>Let <span class="math inline">\(Y_{ij}\)</span> represent the count of interactions between a bait <span class="math inline">\(i\)</span> and a fragment <span class="math inline">\(j\)</span>,<span class="math inline">\(d_{ij}\)</span> be the distance between the bait and the otherEnd and <span class="math inline">\(t_{i}\)</span> denote the number of reads linking a bait <span class="math inline">\(i\)</span> with another fragment in trans. <span class="math inline">\(Y_{ij}\)</span> is such that it is assumed to follow a distribution with a mean <span class="math inline">\(\mu_{ij}\)</span> as: <span class="math display">\[ \mu_{ij} = \beta_0 + \beta_1 \log(d_{ij}) + \beta_2 \log(t_i+1) \]</span></p>
<p>For bait to bait interactions, terms are added to the model to adjust for trans counts of the other end, fragment j, and the product of trans counts of both fragments as: <span class="math display">\[ \mu_{ij} = \beta_0 + \beta_1 \log(d_{ij}) + \beta_2 \log(t_i+1)  + \beta_3 \log(t_j + 1) + \beta_4 \log(t_i +1) \log(t_j+1)\]</span> Each possible interaction is assaigned an expected <span class="math inline">\(\mu_{ij}\)</span> and a p-value for the observed counts <span class="math inline">\(y_{ij}\)</span> versus what is expected by chance is calculated as: <span class="math display">\[p = P(Y_{ij} \geq y_{ij})\]</span></p>
</div>
<div id="score-distribution-for-the-data." class="section level1">
<h1>Score distribution for the data.</h1>
<p>Let us first look at the scores for our data. The threshold value used in the chicane paper is q-value&lt;0.005</p>
<p><img src="index_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
</div>
<div id="cis-interactions" class="section level1">
<h1>Cis-interactions</h1>
<p>Since majority of interactions detected are trans interactions we would like compare how is the distribution of q.value in both the cases.</p>
<div class="figure">
<img src="index_files/figure-html/unnamed-chunk-3-1.png" alt="Distribution of q.value" width="50%" /><img src="index_files/figure-html/unnamed-chunk-3-2.png" alt="Distribution of q.value" width="50%" />
<p class="caption">
Distribution of q.value
</p>
</div>
<p>We can sort the significant interactions using a threshold of 0.05</p>
<p><img src="index_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
Here is a table of the highest q-values observed
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["target.id"],"name":[1],"type":["chr"],"align":["left"]},{"label":["bait.id"],"name":[2],"type":["chr"],"align":["left"]},{"label":["target.chr"],"name":[3],"type":["chr"],"align":["left"]},{"label":["target.start"],"name":[4],"type":["int"],"align":["right"]},{"label":["target.end"],"name":[5],"type":["int"],"align":["right"]},{"label":["bait.chr"],"name":[6],"type":["chr"],"align":["left"]},{"label":["bait.start"],"name":[7],"type":["int"],"align":["right"]},{"label":["bait.end"],"name":[8],"type":["int"],"align":["right"]},{"label":["bait.to.bait"],"name":[9],"type":["lgl"],"align":["right"]},{"label":["bait.trans.count"],"name":[10],"type":["int"],"align":["right"]},{"label":["target.trans.count"],"name":[11],"type":["int"],"align":["right"]},{"label":["distance"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["count"],"name":[13],"type":["int"],"align":["right"]},{"label":["expected"],"name":[14],"type":["dbl"],"align":["right"]},{"label":["p.value"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["q.value"],"name":[16],"type":["dbl"],"align":["right"]},{"label":["bait_name"],"name":[17],"type":["chr"],"align":["left"]}],"data":[{"1":"chr11:65497688-65506516","2":"chr10:75431624-76560168","3":"chr11","4":"65497688","5":"65506516","6":"chr10","7":"75431624","8":"76560168","9":"TRUE","10":"15428","11":"1017967","12":"NA","13":"816","14":"15.098128","15":"8.647309e-195","16":"5.660528e-191","17":"LRMDA"},{"1":"chr11:65497688-65506516","2":"chr10:65912523-67696195","3":"chr11","4":"65497688","5":"65506516","6":"chr10","7":"65912523","8":"67696195","9":"TRUE","10":"8445","11":"1017967","12":"NA","13":"602","14":"10.872046","15":"1.471637e-181","16":"5.904207e-178","17":"CTNNA3"},{"1":"chr11:65497688-65506516","2":"chr10:76869601-77638369","3":"chr11","4":"65497688","5":"65506516","6":"chr10","7":"76869601","8":"77638369","9":"TRUE","10":"9987","11":"1017967","12":"NA","13":"611","14":"11.912425","15":"5.955975e-172","16":"2.740344e-168","17":"KCNMA1"},{"1":"chr11:65497688-65506516","2":"chr1:56994778-58546734","3":"chr11","4":"65497688","5":"65506516","6":"chr1","7":"56994778","8":"58546734","9":"TRUE","10":"11447","11":"1017967","12":"NA","13":"636","14":"12.831882","15":"2.084479e-169","16":"1.314056e-165","17":"DAB1"},{"1":"chr11:65497688-65506516","2":"chr1:6785454-7769706","3":"chr11","4":"65497688","5":"65506516","6":"chr1","7":"6785454","8":"7769706","9":"TRUE","10":"9019","11":"1017967","12":"NA","13":"567","14":"11.268684","15":"1.684781e-165","16":"9.003467e-162","17":"CAMTA1"},{"1":"chr11:65497688-65506516","2":"chr10:53802771-55627942","3":"chr11","4":"65497688","5":"65506516","6":"chr10","7":"53802771","8":"55627942","9":"TRUE","10":"7661","11":"1017967","12":"NA","13":"525","14":"10.309893","15":"6.794550e-163","16":"2.537765e-159","17":"PCDH15"},{"1":"chr11:65497688-65506516","2":"chr1:87620803-88685204","3":"chr11","4":"65497688","5":"65506516","6":"chr1","7":"87620803","8":"88685204","9":"TRUE","10":"15801","11":"1017967","12":"NA","13":"689","14":"15.295962","15":"2.541817e-160","16":"2.060651e-156","17":"PKN2-AS1"},{"1":"chr11:65497688-65506516","2":"chr10:50990888-52298423","3":"chr11","4":"65497688","5":"65506516","6":"chr10","7":"50990888","8":"52298423","9":"TRUE","10":"24078","11":"1017967","12":"NA","13":"818","14":"19.242720","15":"2.793842e-159","16":"2.641298e-155","17":"PRKG1"},{"1":"chr6:63719980-65707226","2":"chr6:63635802-63779336","3":"chr6","4":"63719980","5":"65707226","6":"chr6","7":"63635802","8":"63779336","9":"TRUE","10":"4774","11":"12409","12":"1006034","13":"3938","14":"31.612910","15":"2.607435e-151","16":"2.245001e-148","17":"PHF3"},{"1":"chr11:65497688-65506516","2":"chr1:48532854-50023954","3":"chr11","4":"65497688","5":"65506516","6":"chr1","7":"48532854","8":"50023954","9":"TRUE","10":"7066","11":"1017967","12":"NA","13":"476","14":"9.865561","15":"2.618277e-151","16":"1.075850e-147","17":"AGBL4"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<div id="distribution-of-number-of-reads-wrt-to-distance-and-score" class="section level2">
<h2>Distribution of number of reads wrt to distance and score</h2>
<p>We look at the distribution of read count and scores with the distance of the interaction</p>
<div class="figure">
<img src="index_files/figure-html/unnamed-chunk-6-1.png" alt="Distribution of N-reads" width="50%" /><img src="index_files/figure-html/unnamed-chunk-6-2.png" alt="Distribution of N-reads" width="50%" />
<p class="caption">
Distribution of N-reads
</p>
</div>
</div>
</div>
<div id="score-relationship-with-distance" class="section level1">
<h1>Score relationship with distance</h1>
<p>Another plot to compare is the distribution of the q.value with respect to the distance of the interactions.</p>
<div class="figure">
<img src="index_files/figure-html/unnamed-chunk-7-1.png" alt="Distribution of q.value" width="50%" /><img src="index_files/figure-html/unnamed-chunk-7-2.png" alt="Distribution of q.value" width="50%" />
<p class="caption">
Distribution of q.value
</p>
</div>
</div>
<div id="correlation-plot" class="section level1">
<h1>Correlation plot</h1>
<p>We can plot the correlation between the count of each interaction with the <span class="math inline">\(p\)</span>-value. <img src="index_files/figure-html/unnamed-chunk-8-1.png" width="672" /> # Percentage of genome covered The following plot indicates what fraction of reads for each RNA are on its home chromosome.</p>
<p><img src="index_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<pre><code>##             used   (Mb) gc trigger   (Mb)   max used   (Mb)
## Ncells  11689520  624.3   17671836  943.8   17671836  943.8
## Vcells 798374066 6091.2 1215778551 9275.7 1070454338 8167.0</code></pre>
<div id="trans-vs-cis-interactions" class="section level2">
<h2>Trans vs cis interactions</h2>
<p>It should be noted that majority of the interactions in our data are trans interactions(87.9932091), therefore the distance relation of the score can not be visualized easily here.</p>
<p><img src="index_files/figure-html/r-1.png" width="672" /></p>
</div>
<div id="comparison-with-previous-plots" class="section level2">
<h2>Comparison with previous plots</h2>
<p>Previously we had seen certain plots which stood out when looking at the relationship between the probability of detencting an interaction and the distance.</p>
<p><img src="index_files/figure-html/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="index_files/figure-html/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="we-can-compare-the-plots-for-these-with-plot-of-q.value-vs-distance-for-these-particular-rnas." class="section level3">
<h3>We can compare the plots for these with plot of q.value vs distance for these particular RNAs.</h3>
<p><img src="index_files/figure-html/unnamed-chunk-13-1.png" width="672" /> <img src="index_files/figure-html/unnamed-chunk-14-1.png" width="672" /> We can compare these plots to the cases where significant interactions are found. Such as, <img src="index_files/figure-html/unnamed-chunk-15-1.png" width="672" /> <img src="index_files/figure-html/unnamed-chunk-16-1.png" width="672" /> <img src="index_files/figure-html/unnamed-chunk-17-1.png" width="672" /> <img src="index_files/figure-html/unnamed-chunk-18-1.png" width="672" /> <img src="index_files/figure-html/unnamed-chunk-19-1.png" width="672" /></p>
<p><img src="index_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
