---
title: "Analysis of Chicane Results"
author: "Shashank Tiwari"
date: "Updated<br> `r Sys.Date()`"
output: 
  rmarkdown::html_document: 
    theme: spacelab
    highlight: zenburn 
    code_folding: show 
    toc: true 
    toc_float: true
    smooth_scroll: true
    number_sections: false 
    self_contained: true 
editor_options: 
  chunk_output_type: inline
---

```{r setup, include=T, message=F}
knitr::opts_chunk$set(
  collapse = TRUE,
  dpi=350,
  fig.height = 6,
  fig.width = 7.00787 #178mm
)  
knitr::opts_chunk$set(warning = FALSE)  
#Input
library(knitr)
library(png)
library(gridExtra)
library(ggplot2)
library(genomation)
library(GenomicRanges)
library(data.table)
library(dplyr)
library(stringr)
library(hrbrthemes)
library(viridis)
library(slickR)
library(svglite)
```


# Initial Observations

We look to analyse the ouput from the initial chicane run. Compare the results from the previous Chicago run.






# Introduction to the Chicane 

CHiCANE was designed to identify regions of interest that interact more than expected by chance. To indetify the interaction peaks, CHiCANE models the expected number of reads linking two restriction fragments as a function of the distance between the loci and the ‘interactibility’ of the bait fragment; that is, its inherent propensity to interact with other fragments.

Let $Y_{ij}$ represent the count of interactions between a bait $i$ and a fragment $j$,$d_{ij}$ be the distance between the bait and the otherEnd and $t_{i}$ denote the number of reads linking a bait $i$ with another fragment in trans. $Y_{ij}$ is such that it is assumed to follow a distribution with a mean $\mu_{ij}$ as:
$$ \mu_{ij} = \beta_0 + \beta_1 \log(d_{ij}) + \beta_2 \log(t_i+1) $$


For bait to bait interactions, terms are added to the model to adjust for trans counts of the other end, fragment j, and the product of trans counts of both fragments as:
$$ \mu_{ij} = \beta_0 + \beta_1 \log(d_{ij}) + \beta_2 \log(t_i+1)  + \beta_3 \log(t_j + 1) + \beta_4 \log(t_i +1) \log(t_j+1)$$
Each possible interaction is assaigned an expected $\mu_{ij}$ and a p-value for the observed counts $y_{ij}$ versus what is expected by chance is calculated as:
$$p = P(Y_{ij} \geq y_{ij})$$
# Example data

Let us first look at plots from the example dataset provided with the chicane paper.

```{r}
# Read example data file
ex = fread("/home/master/shared_folder/radicl_seq/Data/Raw/CHiCANE_Supplementary_Data/data/chicane/T47D/interaction_calls.txt.gz",header=TRUE,stringsAsFactors = FALSE,quote = "")
# Significant interactions
ex_signif <- ex[ex$`"q.value"`<0.05]
```

## Score distribution of example data
This is the output for the example data. A threshold of 0.05 is used to determine significant interactions.

```{r}
ggplot(ex,aes(ex$`"q.value"`))+
  geom_histogram(fill="#69b3a2", color="#e9ecef",alpha=0.8)+
  ggtitle("Score distribution for example data")+
  theme_ipsum()+
  theme(plot.title = element_text(size = 10))
```
### Cis vs Trans interactions

```{r,fig.show="hold",out.width="50%",fig.cap="Distribution of q.value",message=FALSE}
##Cis-trans-ex_plot
par(mar=c(4,4,.1,.1))


ggplot(ex[!is.na(ex$`"distance"`)],aes(ex[!is.na(ex$`"distance"`)]$`"q.value"`))+
  geom_histogram(fill="#69b3a2", color="#e9ecef",alpha=0.8)+
  ggtitle("Score distribution for example data(Cis)")+
  theme_ipsum()+
  theme(plot.title = element_text(size = 10))+
  xlab("q.value")

ggplot(ex[is.na(ex$`"distance"`)],aes(ex[is.na(ex$`"distance"`)]$`"q.value"`))+
  geom_histogram(fill="#69b3a2", color="#e9ecef",alpha=0.8)+
  ggtitle("Score distribution for example data(Trans)")+
  theme_ipsum()+
  theme(plot.title = element_text(size = 10))+
  xlab("q.value")
```

```{r,echo=FALSE,message=FALSE,results='hide'}
gc()
```

## Distance distibution for example data
We look at the counts of distance for the interactions.

```{r}
ggplot(ex,aes(ex$`"distance"`))+
  geom_histogram(fill="#69b3a2", color="#e9ecef",alpha=0.8)+
  ggtitle("Distribution of distance for example data")+
  theme_ipsum()+
  theme(plot.title = element_text(size = 10))+
  xlab("distance")
```

### Significant interactions
Filtering this data for the significant interactions.
```{r}
ggplot(ex_signif,aes(ex_signif$`"distance"`))+
  geom_histogram(fill="#69b3a2", color="#e9ecef",alpha=0.8)+
  ggtitle("Distribution of distance for example data")+
  theme_ipsum()+
  theme(plot.title = element_text(size = 10))+
  xlab("distance")
```

## Correlation plot
We plot the correlation between the observed p.value and the counts of interaction for example data. Note that the p.value here is unweighted.


```{r,message=FALSE,fig.show="hold",out.width="50%",fig.cap="Cis vs Trans"}
par(mar=c(4,4,.1,.1))

cor_test <- cor.test(ex[is.na(ex$`"distance"`)]$`"count"`,ex[is.na(ex$`"distance"`)]$`"p.value"`,method = "spearman",exact = F)
plt_text <- paste0('r = ', round(cor_test$estimate,3), ', ',
                paste('p-value < 2.2e-16'))#taken from res of cor_test
setorder(ex[is.na(ex$`"distance"`)],`"distance"`)
ggplot(data=ex[is.na(ex$`"distance"`)],aes(x = log(`"count"`,base=10), 
                      y = log(`"p.value"`,base=10),colour=`"distance"`))+ 
  geom_point(alpha=0.2,size=0.1) + 
  geom_smooth(method = "lm", se = T,color = 'grey20', alpha = 0.2,size=0.5) +
  annotate('text', x = -Inf, y = Inf, hjust = -.1, vjust = 1  ,label=plt_text)+
  labs(y= "Log 10 P-Value", x = "Log 10 Interaction Counts(Trans)")  +
  theme_ipsum()+
  ggtitle("Example Data-Trans inetractions")+
  scale_color_viridis()

cor_test <- cor.test(ex[!is.na(ex$`"distance"`)]$`"count"`,ex[!is.na(ex$`"distance"`)]$`"p.value"`,method = "spearman",exact = F)
plt_text <- paste0('r = ', round(cor_test$estimate,3), ', ',
                paste('p-value < 2.2e-16'))#taken from res of cor_test
setorder(ex[!is.na(ex$`"distance"`)],`"distance"`)
ggplot(ex[!is.na(ex$`"distance"`)] ,aes(x = log(`"count"`,base=10), 
                      y = log(`"p.value"`,base=10),colour=`"distance"`))+ 
  geom_point(alpha=0.2,size=0.1) + 
  geom_smooth(method = "lm", se = T,color = 'grey20', alpha = 0.2,size=0.5) +
  annotate('text', x = -Inf, y = Inf, hjust = -.1, vjust = 1  ,label=plt_text)+
  labs(y= "Log 10 P-Value", x = "Log 10 Interaction Counts(Cis)")  +
  theme_ipsum()+
  ggtitle("Example Data-Cis inetractions")+
  scale_color_viridis()

```

```{r,echo=FALSE,message=FALSE,results='hide'}
rm(ex)
gc()
```

# Radicl-seq data


```{r}

#Read output data file
ibed <- fread("/home/master/radicl_seq/Run_chicago/Data/chicane.results.rdcl.txt",header=TRUE,stringsAsFactors = FALSE,quote = "")
#Remove NA entries
ibed <- ibed[!is.na(ibed$p.value)]

#Read transcript file
baits <- fread("/home/master/shared_folder/radicl_seq/Data/Raw/all_genes_location.txt",header = FALSE,stringsAsFactors = FALSE)
colnames(baits) <- c("chr","start","end","bait_name")
baits$mid <- 0.5*(baits$end-baits$start)
ibed$bait_name <- baits$bait_name[match(ibed$bait.start,baits$start)]

```

## Score distribution for radicl-seq.

Let us first look at the scores for radicl-seq.
```{r}
ggplot(ibed,aes(q.value))+
  geom_histogram(fill="#69b3a2", color="#e9ecef",alpha=0.8)+
  ggtitle("Score distribution for radicl-seq")+
  theme_ipsum()
```

### Cis vs trans interactions

Since majority of interactions detected are trans interactions we would like compare how is the distribution of q.value in both the cases.

```{r,fig.show="hold",out.width="50%",fig.cap="Distribution of q.value"}
##Cis-trans-ex_plot
par(mar=c(4,4,.1,.1))
ggplot(ibed[!is.na(ibed$distance)],aes(q.value))+
  geom_histogram(fill="#69b3a2", color="#e9ecef",alpha=0.8)+
  ggtitle("Score distribution for data(Cis)")+
  theme_ipsum()

ggplot(ibed[is.na(ibed$distance)],aes(q.value))+
  geom_histogram(fill="#69b3a2", color="#e9ecef", alpha=0.8)+
  ggtitle("Score distribution for data(Trans)")+
  theme_ipsum()
```

```{r,echo=FALSE,message=FALSE,results='hide'}
gc()
```

## Distance distribution

```{r}
ggplot(ibed,aes(distance))+
  geom_histogram(fill="#69b3a2", color="#e9ecef",alpha=0.8)+
  ggtitle("Distribution of distance for radicl-seq")+
  theme_ipsum()+
  theme(plot.title = element_text(size = 10))
```

### Significant interactions
Filtering this data for the significant interactions.
```{r}

signif <-ibed[ibed$q.value<0.05,]

ggplot(signif,aes(distance))+
  geom_histogram(fill="#69b3a2", color="#e9ecef",alpha=0.8)+
  ggtitle("Distribution of distance for radicl-seq")+
  theme_ipsum()+
  theme(plot.title = element_text(size = 10))
```

## Distance relation of q.value

Next we look at the distribution of the q,value with the distance and the counts.

```{r}


p1 = ggplot(signif %>% arrange(count),aes(x=distance,y=log(q.value,base=10),color=log(count,base=10)))+
      geom_point(size=0.5,alpha=0.5)+
      theme_ipsum()+
      scale_color_gradient(high="red",low="#69b3a2",name="log.count")+
    xlab("distance")+
  ylab("log.q.value")+
  ggtitle("q.value distribution of significant interactions for radicl-seq")+
  theme(plot.title = element_text(size = 10))

p2 = ggplot(ex_signif %>% arrange(`"count"`),aes(x=ex_signif$`"distance"` ,y=log(ex_signif$`"q.value"`,base=10),color=log(ex_signif$`"count"`,base=10)))+
    scale_color_gradient(high="red",low="#69b3a2",name="log.count")+
   geom_point(size=0.5,alpha=0.5)+
    theme_ipsum()+
  xlab("distance")+
  ylab("log.q.value")+
  ggtitle("q.value distribution of significant interactions for example data")+
  theme(plot.title = element_text(size = 10))

p1 = xmlSVG({show(p1)}, standalone = TRUE)  
p2 = xmlSVG({show(p2)}, standalone = TRUE)  
slickR(list(p1, p2)) + settings(dots = TRUE, autoplay = TRUE)
```

```{r,echo=FALSE,message=FALSE,results='hide'}
gc()
```

Here is a table of the highest q-values observed
```{r}
signif <- signif[with(signif,order(q.value)),]
top  <- signif[1:10,]
print(top)
```



```{r,echo=FALSE,message=FALSE,results='hide'}
gc()
```



## Correlation plot

We can plot the correlation between the count of each interaction with the $p$-value. 


```{r,message=FALSE}

cor_test <- cor.test(ibed$count,ibed$p.value,method = "spearman",exact = F)
plt_text <- paste0('r = ', round(cor_test$estimate,3), ', ',
                paste('p-value < 2.2e-16'))#taken from res of cor_test
setorder(ibed,distance)
ggplot(data=ibed,aes(x = log(count,base=10), 
                      y = log(p.value,base=10),colour=distance))+ 
  geom_point(alpha=0.2,size=0.1) + 
  geom_smooth(method = "lm", se = T,color = 'grey20', alpha = 0.2,size=0.5) +
  annotate('text', x = -Inf, y = Inf, hjust = -.1, vjust = 1  ,label=plt_text)+
  labs(y= "Log 10 P-Value", x = "Log 10 Interaction Counts")  +
  theme_ipsum()+
  scale_color_viridis()
```

```{r,echo=FALSE,message=FALSE,results='hide'}
gc()
```

Splitting this plot for cis and trans interactions

```{r,message=FALSE,fig.show="hold",out.width="50%",fig.cap="Cis vs Trans"}
par(mar=c(4,4,.1,.1))
cor_test <- cor.test(ibed[is.na(ibed$distance)]$count,ibed[is.na(ibed$distance)]$p.value,method = "spearman",exact = F)
plt_text <- paste0('r = ', round(cor_test$estimate,3), ', ',
                paste('p-value < 2.2e-16'))#taken from res of cor_test
setorder(ibed[is.na(ibed$distance)],distance)
ggplot(data=ibed[is.na(ibed$distance)],aes(x = log(count,base=10), 
                      y = log(p.value,base=10),colour=distance))+ 
  geom_point(alpha=0.2,size=0.1) + 
  geom_smooth(method = "lm", se = T,color = 'grey20', alpha = 0.2,size=0.5) +
  annotate('text', x = -Inf, y = Inf, hjust = -.1, vjust = 1  ,label=plt_text)+
  labs(y= "Log 10 P-Value", x = "Log 10 Interaction Counts(Trans)")  +
  theme_ipsum()+
  ggtitle("Trans inetractions")+
  scale_color_viridis()

cor_test <- cor.test(ibed[!is.na(ibed$distance)]$count,ibed[!is.na(ibed$distance)]$p.value,method = "spearman",exact = F)
plt_text <- paste0('r = ', round(cor_test$estimate,3), ', ',
                paste('p-value < 2.2e-16'))#taken from res of cor_test
setorder(ibed[!is.na(ibed$distance)],distance)
ggplot(ibed[!is.na(ibed$distance)] %>% arrange(distance),aes(x = log(count,base=10), 
                      y = log(p.value,base=10),colour=distance))+ 
  geom_point(alpha=0.2,size=0.1) + 
  geom_smooth(method = "lm", se = T,color = 'grey20', alpha = 0.2,size=0.5) +
  annotate('text', x = -Inf, y = Inf, hjust = -.1, vjust = 1  ,label=plt_text)+
  labs(y= "Log 10 P-Value", x = "Log 10 Interaction Counts(Cis)")  +
  theme_ipsum()+
  ggtitle("Cis inetractions")+
  scale_color_viridis()
```

```{r,echo=FALSE,message=FALSE,results='hide'}
gc()
```





## q.value vs count

It should be noted that majority of the interactions in radicl-seq are trans interactions(`r length(which(is.na(ibed$distance)))*100/length(ibed$bait.id)`), therefore the distance relation of the score can not be visualized easily here. 


```{r ,message=FALSE, warning=FALSE, r, }
ggplot(ibed[is.na(ibed$distance)],aes(count,log(q.value,base=10)))+
  geom_point(color="#69b3a2",size=0.5,alpha=0.3)+
  theme_ipsum()+
  ggtitle("Count distribution of trans interaction")
```

# Comparison with previous plots

Previously we had seen certain plots which stood out when looking at the relationship between the probability of detecting an interaction and the distance. We compare these plots with the current q.value distribution.

## MALAT1

```{r}
#MALAT1

# read radicl-seq data bedpe file
yyy = fread("/home/master/shared_folder/radicl_seq/Data/Raw/Day13_done.bedpe",header=FALSE,stringsAsFactors = FALSE,quote = "")
colnames(yyy) <- c('rnachrom','rnachromStart','rnachromEnd','dnachrom','dnachromStart','dnachromEnd','rnaID')
yyy$bait_mid <- baits$mid[match(yyy$rnaID,baits$bait_name)]
yyy$len <- abs((yyy$dnachromStart+yyy$dnachromEnd)/2-yyy$bait_mid)
yyy$len <- ifelse(yyy$rnachrom==yyy$dnachrom,yyy$len,NA)

MALAT1 <- ibed[ibed$bait_name=="MALAT1"]
q.count <- quantile(MALAT1$count,c(0.30,0.60,0.90,0.95))
MALAT1$q.count <- with(MALAT1,ifelse(count<q.count[1],0,
                                 ifelse(count<q.count[2],1,
                                  ifelse(count<q.count[3],2,
                                  ifelse(count<q.count[4],3,4)))))

#dist_malat
p1 = ggplot(yyy[yyy$rnaID=="MALAT1"],aes(x = len))+
  geom_histogram(fill="#69b3a2",aes(y = stat(count)/sum(count)))+
  scale_y_continuous(labels = scales::percent)+
  ggtitle("MALAT1")+
  xlab("Distance of interaction")+
  ylab("Probability")+
  theme_ipsum()
setorder(MALAT1,q.count)
p2 = ggplot(MALAT1 ,aes(x=distance,y=log(q.value,base=10)))+
    geom_point(aes(color=as.factor(q.count)),size=0.5,alpha=0.3)+
    scale_colour_manual(name="percentile of counts",
                    values=c("#69b3a2","black","darkgoldenrod4","coral1","red"),labels=c("0-30","30-60","60-90","90-95","95-100"))+
    ggtitle("MALAT1")+
    theme_ipsum()+
    ylab("log(q.value)")

p1 = xmlSVG({show(p1)}, standalone = TRUE)  
p2 = xmlSVG({show(p2)}, standalone = TRUE)  
slickR(list(p1, p2)) + settings(dots = TRUE, autoplay = TRUE)
```

## RNVU1-7

```{r}
#RNVU1-7
RNVU1 <- ibed[ibed$bait_name=="RNVU1-7"]
q.count <- quantile(RNVU1$count,c(0.30,0.60,0.90,0.95))
RNVU1$q.count <- with(RNVU1,ifelse(count<q.count[1],0,
                                 ifelse(count<q.count[2],1,
                                  ifelse(count<q.count[3],2,
                                  ifelse(count<q.count[4],3,4)))))


p1 = ggplot(yyy[yyy$rnaID=="RNVU1-7"],aes(x = len))+
  geom_histogram(fill="#69b3a2",aes(y = stat(count)/sum(count)))+
  scale_y_continuous(labels = scales::percent)+
  ggtitle("RNVU1-7")+
  xlab("Distance of interaction")+
  ylab("Probability")+
  theme_ipsum()
setorder(RNVU1,q.count)
p2 = ggplot(RNVU1 ,aes(x=distance,y=log(q.value,base=10)))+
    geom_point(aes(color=as.factor(q.count)),size=0.5,alpha=0.3)+
    scale_colour_manual(name="percentile of counts",
                    values=c("#69b3a2","red"),labels=c("60-90","95-100"))+
    ggtitle("RNVU1-7")+
    theme_ipsum()+
    ylab("log(q.value)")

p1 = xmlSVG({show(p1)}, standalone = TRUE)  
p2 = xmlSVG({show(p2)}, standalone = TRUE)  
slickR(list(p1, p2)) + settings(dots = TRUE, autoplay = TRUE)  
```


## RMRP

```{r}
#RMRP
RMRP <- ibed[ibed$bait_name=="RMRP"]
q.count <- quantile(RMRP$count,c(0.30,0.60,0.90,0.95))
RMRP$q.count <- with(RMRP,ifelse(count<q.count[1],0,
                                 ifelse(count<q.count[2],1,
                                  ifelse(count<q.count[3],2,
                                  ifelse(count<q.count[4],3,4)))))

#dist_malat
p1 = ggplot(yyy[yyy$rnaID=="RMRP"],aes(x = len))+
  geom_histogram(fill="#69b3a2",aes(y = stat(count)/sum(count)))+
  scale_y_continuous(labels = scales::percent)+
  ggtitle("RMRP")+
  xlab("Distance of interaction")+
  ylab("Probability")+
  theme_ipsum()
setorder(RMRP,q.count)
p2 = ggplot(RMRP,aes(x=distance,y=log(q.value,base=10)))+
    geom_point(aes(color=as.factor(q.count)),size=0.5,alpha=0.3)+
    scale_colour_manual(name="percentile of counts",
                    values=c("darkgoldenrod4","coral1","red"),labels=c("60-90","90-95","95-100"))+
    ggtitle("RMRP")+
    theme_ipsum()+
    ylab("log(q.value)")

p1 = xmlSVG({show(p1)}, standalone = TRUE)  
p2 = xmlSVG({show(p2)}, standalone = TRUE)  
slickR(list(p1, p2)) + settings(dots = TRUE, autoplay = TRUE) 
```
## EBF1

```{r}
#EBF1
EBF1 <- ibed[ibed$bait_name=="EBF1"]
q.count <- quantile(EBF1$count,c(0.30,0.60,0.90,0.95))
EBF1$q.count <- with(EBF1,ifelse(count<q.count[1],0,
                                 ifelse(count<q.count[2],1,
                                  ifelse(count<q.count[3],2,
                                  ifelse(count<q.count[4],3,4)))))

#dist_malat
p1 = ggplot(yyy[yyy$rnaID=="EBF1"],aes(x = len))+
  geom_histogram(fill="#69b3a2",aes(y = stat(count)/sum(count)))+
  scale_y_continuous(labels = scales::percent)+
  ggtitle("EBF1")+
  xlab("Distance of interaction")+
  ylab("Probability")+
  theme_ipsum()

setorder(EBF1,q.count)
p2 = ggplot(EBF1,aes(x=distance,y=log(q.value,base=10)))+
    geom_point(aes(color=as.factor(q.count)),size=0.5,alpha=0.3)+
    scale_colour_manual(name="percentile of counts",
                    values=c("black","darkgoldenrod4","coral1","red"),labels=c("30-60","60-90","90-95","95-100"))+ 
    ggtitle("EBF1")+
    theme_ipsum()+
    ylab("log(q.value)")

p1 = xmlSVG({show(p1)}, standalone = TRUE)  
p2 = xmlSVG({show(p2)}, standalone = TRUE)  
slickR(list(p1, p2)) + settings(dots = TRUE, autoplay = TRUE)
```
## TENM4

```{r}
#TENM4
TENM4 <- ibed[ibed$bait_name=="TENM4"]
q.count <- quantile(TENM4$count,c(0.30,0.60,0.90,0.95))
TENM4$q.count <- with(TENM4,ifelse(count<q.count[1],0,
                                 ifelse(count<q.count[2],1,
                                  ifelse(count<q.count[3],2,
                                  ifelse(count<q.count[4],3,4)))))

#dist_malat
p1 = ggplot(yyy[yyy$rnaID=="TENM4"],aes(x = len))+
  geom_histogram(fill="#69b3a2",aes(y = stat(count)/sum(count)))+
  scale_y_continuous(labels = scales::percent)+
  ggtitle("TENM4")+
  xlab("Distance of interaction")+
  ylab("Probability")+
  theme_ipsum()

setorder(TENM4,q.count)
p2 = ggplot(TENM4,aes(x=distance,y=log(q.value,base=10)))+
    geom_point(aes(color=as.factor(q.count)),size=0.5,alpha=0.3)+
    scale_colour_manual(name="percentile of counts",
                    values=c("black","darkgoldenrod4","coral1","red"),labels=c("30-60","60-90","90-95","95-100"))+ 
    ggtitle("TENM4")+
    theme_ipsum()+
    ylab("log(q.value)")

p1 = xmlSVG({show(p1)}, standalone = TRUE)  
p2 = xmlSVG({show(p2)}, standalone = TRUE)  
slickR(list(p1, p2)) + settings(dots = TRUE, autoplay = TRUE)
```

## NEAT1

```{r}
#NEAT1
NEAT1 <- ibed[ibed$bait_name=="NEAT1"]
q.count <- quantile(NEAT1$count,c(0.30,0.60,0.90,0.95))
NEAT1$q.count <- with(NEAT1,ifelse(count<q.count[1],0,
                                 ifelse(count<q.count[2],1,
                                  ifelse(count<q.count[3],2,
                                  ifelse(count<q.count[4],3,4)))))

#dist_malat
p1 = ggplot(yyy[yyy$rnaID=="NEAT1"],aes(x = len))+
  geom_histogram(fill="#69b3a2",aes(y = stat(count)/sum(count)))+
  scale_y_continuous(labels = scales::percent)+
  ggtitle("NEAT1")+
  xlab("Distance of interaction")+
  ylab("Probability")+
  theme_ipsum()

setorder(NEAT1,q.count)
p2 = ggplot(NEAT1,aes(x=distance,y=log(q.value,base=10)))+
    geom_point(aes(color=as.factor(q.count)),size=0.5,alpha=0.3)+
    scale_colour_manual(name="percentile of counts",
                    values=c("#69b3a2","black","darkgoldenrod4","coral1","red"),labels=c("0-30","30-60","60-90","90-95","95-100"))+ 
    ggtitle("NEAT1")+
    theme_ipsum()+
    ylab("log(q.value)")

p1 = xmlSVG({show(p1)}, standalone = TRUE)  
p2 = xmlSVG({show(p2)}, standalone = TRUE)  
slickR(list(p1, p2)) + settings(dots = TRUE, autoplay = TRUE)
```

## EHBP1

```{r}
#EHBP1
EHBP1 <- ibed[ibed$bait_name=="EHBP1"]
q.count <- quantile(EHBP1$count,c(0.30,0.60,0.90,0.95))
EHBP1$q.count <- with(EHBP1,ifelse(count<q.count[1],0,
                                 ifelse(count<q.count[2],1,
                                  ifelse(count<q.count[3],2,
                                  ifelse(count<q.count[4],3,4)))))

#dist_malat
p1 = ggplot(yyy[yyy$rnaID=="EHBP1"],aes(x = len))+
  geom_histogram(fill="#69b3a2",aes(y = stat(count)/sum(count)))+
  scale_y_continuous(labels = scales::percent)+
  ggtitle("EHBP1")+
  xlab("Distance of interaction")+
  ylab("Probability")+
  theme_ipsum()

setorder(EHBP1,q.count)
p2 = ggplot(EHBP1,aes(x=distance,y=log(q.value,base=10)))+
    geom_point(aes(color=as.factor(q.count)),size=0.5,alpha=0.3)+
    scale_colour_manual(name="percentile of counts",
                    values=c("black","darkgoldenrod4","coral1","red"),labels=c("30-60","60-90","90-95","95-100"))+ 
    ggtitle("EHBP1")+
    theme_ipsum()+
    ylab("log(q.value)")

p1 = xmlSVG({show(p1)}, standalone = TRUE)  
p2 = xmlSVG({show(p2)}, standalone = TRUE)  
slickR(list(p1, p2)) + settings(dots = TRUE, autoplay = TRUE)  
```

## PDE3B

```{r}
#PDE3B
PDE3B <- ibed[ibed$bait_name=="PDE3B"]
q.count <- quantile(PDE3B$count,c(0.30,0.60,0.90,0.95))
PDE3B$q.count <- with(PDE3B,ifelse(count<q.count[1],0,
                                 ifelse(count<q.count[2],1,
                                  ifelse(count<q.count[3],2,
                                  ifelse(count<q.count[4],3,4)))))

#dist_malat
p1 = ggplot(yyy[yyy$rnaID=="PDE3B"],aes(x = len))+
  geom_histogram(fill="#69b3a2",aes(y = stat(count)/sum(count)))+
  scale_y_continuous(labels = scales::percent)+
  ggtitle("PDE3B")+
  xlab("Distance of interaction")+
  ylab("Probability")+
  theme_ipsum()

setorder(PDE3B,q.count)
p2 =ggplot(PDE3B,aes(x=distance,y=log(q.value,base=10)))+
    geom_point(aes(color=as.factor(q.count)),size=0.5,alpha=0.3)+
    scale_colour_manual(name="percentile of counts",
                    values=c("black","darkgoldenrod4","green","red"),labels=c("30-60","60-90","90-95","95-100"))+ 
    ggtitle("PDE3B")+
    theme_ipsum()+
    ylab("log(q.value)")

p1 = xmlSVG({show(p1)}, standalone = TRUE)  
p2 = xmlSVG({show(p2)}, standalone = TRUE)  
slickR(list(p1, p2)) + settings(dots = TRUE, autoplay = TRUE)
```
## DAB1

```{r,fig.show="hold",out.width="50%",fig.cap="DAB1 "}
#DAB1

DAB1 <- ibed[ibed$bait_name=="DAB1"]
q.count <- quantile(DAB1$count,c(0.30,0.60,0.90,0.95))
DAB1$q.count <- with(DAB1,ifelse(count<q.count[1],0,
                                 ifelse(count<q.count[2],1,
                                  ifelse(count<q.count[3],2,
                                  ifelse(count<q.count[4],3,4)))))

#dist_malat
p1 = ggplot(yyy[yyy$rnaID=="DAB1"],aes(x = len))+
  geom_histogram(fill="#69b3a2",aes(y = stat(count)/sum(count)))+
  scale_y_continuous(labels = scales::percent)+
  ggtitle("DAB1")+
  xlab("Distance of interaction")+
  ylab("Probability")+
  theme_ipsum()

setorder(DAB1,q.count)
p2 = ggplot(DAB1,aes(x=distance,y=log(q.value,base=10)))+
    geom_point(aes(color=as.factor(q.count)),size=0.5,alpha=0.3)+
    scale_colour_manual(name="percentile of counts",
                    values=c("darkgoldenrod4","green","red"),labels=c("60-90","90-95","95-100"))+ 
    ggtitle("DAB1")+
    theme_ipsum()+
    ylab("log(q.value)")

p1 = xmlSVG({show(p1)}, standalone = TRUE)  
p2 = xmlSVG({show(p2)}, standalone = TRUE)  
slickR(list(p1, p2)) + settings(dots = TRUE, autoplay = TRUE)
```


